### ISSUE TITLE
Loose dependencies of tests on "unzip -t"

### ISSUE THREAD

**Issue by farblos** [redhotpenguin/perl-Archive-Zip#67]
Most of the failures on cpantesters are related to "unzip -t" not working as expected on some platforms.

I'm currently trying to fix that in some elegant, generic way, so this is more of a "heads-up-issue" that there is something in the pipeline.

**Comment by farblos** [comment]
The main idea is to provide a function `azuztok` in `common.pm` that does all that `unzip -t` invocation, output logging, exit value handling, in one call which directly generates TAP  output.

Plus there is one optional parameter `$refzipfile`
```
  azuztok( $file, 'refzip' => $refzipfile )
```
that  allows to specify a reference zip file. If that is specified, `unzip -t` on `$file` must not return exit value 0, but the same exit value as `unzip -t` on `$refzipfile`.

Prototype-wise this works already, but this still needs some brusing up...

**Comment by farblos** [comment]
@redhotpenguin, I'd appreciate if you could delay any white space or other plain clean-up PRs on directory "t" until I'm done with my changes. Getting through that merge hassle just because of ... white space feels, well, wrong. Of course I learn something about git, but that's not my main target right now. Thanks!

**Comment by redhotpenguin** [comment]
Sounds good

**Comment by farblos** [comment]
@redhotpenguin, @pmqs, on my fork, branch [test-redesign](https://github.com/farblos/perl-Archive-Zip/tree/test-redesign), there is a candidate that runs through Travis and AppVeyor, but which still could do with some brushing up and all.  Before I go into that, is there any option to get this tested by cpantesters without having a full-blown release?

**Comment by pmqs** [comment]
@farblos the Travis & AppVeyor platforms seem to be well-behaved as far as `unzip -t` is concerned and that is also the case for the majority of the cpantesters boxes.

The issue on cpantesters is likely to be with the environments that are currently failing. So in a sense if we spin a new release you are likely to either fix the issue partially/completely or not make any difference. 

Assuming this goes to cpantesters, and there are still failures, is there enough diagnostic output to root-cause any issues?

**Comment by farblos** [comment]
@pmqs The challenge with diagnostics is a) to write enough information to diagnose all CPAN tester issues "by just looking at the test results" but b) not to frighten or overwhelm regular users doing a `make test` with copious output.

I resolved to write more or less diagnostics depending on a constant

```
use constant _IN_CPANTESTER_ENVIRONMENT =>
    exists($ENV{'AUTOMATED_TESTING'}) ||
    exists($ENV{'NONINTERACTIVE_TESTING'}) ||
    exists($ENV{'PERL_CPAN_REPORTER_CONFIG'});
```

According to one of the consensus, testing the first environment variable should be sufficient, but the test results on CPAN testers do not list that, but only the 2nd or 3rd.

What do you think?

**Comment by pmqs** [comment]
Hey @farblos. Wasn't aware of those env variables, but if they can help, then sure, go for it.

**Comment by farblos** [comment]
Should be fixed with Archive::Zip version 1.68.
